schema.prisma

```diff
@@ -0,0 +1,682 @@
+  // 必選修，當 `type === Normal` 時才使用，來自舊系統 `sel_code` and `co_rep`
+  compulsory      CourseCompulsory
+  /// 全學年制, 舊系統為1 為全學年制，2為半學年制 (forh)
+  length          CourseLength
+  /// 學分
+  credits         Decimal
+  /// 通識課程配有TA，非通識課程為null，來自舊系統NOL table com
+  isTaAssigned    Boolean?
+  /// 英語授課，來自舊系統NOL table english
+  isEnglishTaught Boolean
+  /// 可充當外文領域
+  asForeign       Boolean?
+  /// 課程大綱網址，來自舊系統 NOL table `ntu_course`
+  url             String?          @db.VarChar(200)
+  /// 密集課程的週次
+  intensiveWeeks  Int[]            @default([])
+
+  // TODO: 完成以下 column 命名
+  studentNumber Int
+  // 外系人數 Int
+  // 外校人數 Int
+  // 加選人數 Int
+
+  // TODO: 是否保留以下 column?
+  /// 實習分組, 0~4
+  groupId                Int // grpno
+  /// 系所縮寫，舊系統時用於產生課號，來自 `dpt_abbr`
+  departmentAbbreviation String @db.VarChar(6)
+  /// 課程流水號，舊系統時用於產生課號，來自 `cou_teacno`
+  codeSerial             String @db.VarChar(4)
+
+  // 1-n
+  practiceGroup PracticeGroup[]
+  schedules     Schedule[]
+  limits        Limit[]
+
+  // n-1
+  gradeId            String
+  /// 開課對象年級，系所查詢時使用，來自舊系統 `year`
+  grade              Grade      @relation(fields: [gradeId], references: [id])
+  targetDepartmentId String
+  targetDepartment   Department @relation(fields: [targetDepartmentId], references: [id])
+  /// 教師代碼
+  teacherId          String
+  teacher            Teacher    @relation(fields: [teacherId], references: [id])
+
+  // m-n
+  prerequisiteOf Course[] @relation("coursePrerequisite")
+  prerequisites  Course[] @relation("coursePrerequisite")
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  translations CourseTranslations[]
+
+  @@unique([semester, serial])
+}
+
+model CourseTranslations {
+  language Language
+
+  /// 課程名稱
+  name    String
+  /// 課程備註
+  comment String
+
+  courseId String @db.Uuid
+  course   Course @relation(fields: [courseId], references: [id])
+
+  @@id([language, courseId])
+}
+
+model Schedule {
+  /// 1~7: 星期一到星期日
+  weekday       Int
+  lessons       String
+  classroomId String @db.Uuid
+  classroom     Classroom @relation(fields: [classroomId], references: [id])
+
+  changed CourseChanged @default(Normal)
+
+  courseId String @db.Uuid
+  course   Course @relation(fields: [courseId], references: [id])
+
+  @@id([courseId, weekday])
+}
+
+model Department {
+  // 系所代碼，來自舊系統 `dpt_code`
+  id       String @id @db.VarChar(4)
+  /// 人事單位代碼，來自舊系統 NOL table `dep_unit.adcd`
+  hrSerial String @db.VarChar(7)
+
+  grades  Grade[]
+  courses Course[]
+
+  translations DepartmentTranslations[]
+}
+
+model DepartmentTranslations {
+  language Language
+
+  /// 系所名稱
+  name      String
+  /// 系所簡稱
+  shortName String
+
+  departmentId String     @db.VarChar(4)
+  department   Department @relation(fields: [departmentId], references: [id])
+
+  @@id([language, departmentId])
+}
+
+/// 計中認證後取得5碼人員編號轉換教務處使用之6碼教師代碼用，來自舊系統 NOL table `tea_emp`, 
+model Teacher {
+  /// 教師代碼，來自舊系統 `tea_code2`
+  id         String @id @db.VarChar(6)
```

Comment:
I used `VarChar` to ensure consistency since the length should always be 6.
What kind of unexpected string do you have in mind? The unexpected strings should be handled before inserting them into DB IMO.
Please correct me if I took this wrong.

Reply:
Normally, we will validate data before inserting them into the database. So we will handle it before the DB validation. 

It's still a good idea to have a DB-level validation in case of some missing checking. But the thing is - if we are using DB-level checking, we will need to handle them in the `catch` section, even needing to parse the error message that returns from DB, which is complex and not easily maintainable.

Reply:
I see. I'll remove all `@db.VarChar` if @jc-hiroto and @Klay24-Huang are good for that.